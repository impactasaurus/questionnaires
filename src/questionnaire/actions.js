import { fetchJson } from '../utils';

export const REQUEST_QUESTIONNAIRE = 'REQUEST_QUESTIONNAIRE';
export const RECEIVE_QUESTIONNAIRE = 'RECEIVE_QUESTIONNAIRE';
export const ERROR_QUESTIONNAIRE = 'ERROR_QUESTIONNAIRE';

export const requestQuestionnaire = () => ({
    type: REQUEST_QUESTIONNAIRE,
});

export const receiveQuestionnaire = questionnaire => ({
    type: RECEIVE_QUESTIONNAIRE,
    questionnaire,
});

export const errorQuestionnaire = error => ({
    type: ERROR_QUESTIONNAIRE,
    error,
});

export const loadQuestionnaire = questionnaire => (dispatch, getState) => {
    const { currentQuestionnaireId, currentQuestionnaire } = getState();
    if (currentQuestionnaireId === questionnaire && currentQuestionnaire) {
        // Already cached
        return;
    }

    dispatch(requestQuestionnaire());
    return fetchJson(`/api/questionnaire/${questionnaire}.json`)
        .then(r => dispatch(receiveQuestionnaire(r)))
        .catch(e => dispatch(errorQuestionnaire(e)));
}