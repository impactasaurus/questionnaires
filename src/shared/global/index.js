import { injectGlobal } from 'styled-components';
import styledReset from 'styled-reset';

export default () => injectGlobal`
    @import url("https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,300italic,400,400italic");
    ${styledReset}

    * {
        box-sizing: border-box;
    }
`;