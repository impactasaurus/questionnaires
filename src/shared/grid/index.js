import R from './Row';
import C from './Column';

export const Row = R;
export const Column = C;