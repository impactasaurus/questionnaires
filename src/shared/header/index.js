import Header from './Header';
import Sub from './Subheader';

export const Subheader = Sub;

export default Header;