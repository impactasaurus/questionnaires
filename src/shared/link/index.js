import Link, { BlankLink as RawBlankLink } from './Link';

export const BlankLink = RawBlankLink;
export default Link;