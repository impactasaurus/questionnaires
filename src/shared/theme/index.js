import rawTypeset from './typeset';
import rawColours from './colours';
import rawSpacings from './spacings';

export const typeset = rawTypeset
export const colours = rawColours;
export const spacings = rawSpacings;

export const primaryTheme = {
    fg: colours.gray60,
    bg: 'transparent',
    highlight: colours.green50,
    lowlight: colours.white,
    gradient: colours.primaryGradient,
};

export const secondaryTheme = {
    fg: colours.white,
    bg: colours.blue30,
    highlight: colours.white,
    lowlight: 'transparent',
    gradient: colours.primaryGradient,
};