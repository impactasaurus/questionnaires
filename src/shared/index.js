import RawAction from './actions';
import RawButton from './button';
import RawHeader, { Subheader as Subhead } from './header';
import RawHero from './hero';
import RawLink, {BlankLink as BLink } from './link';
import RawSection from './section';
import RawTable, { VerticalTable as VTable, TableHeader as TH, TableBody as TB } from './table';
import RawList, { ListItem as LI } from './list';
import { Row as R, Column as C } from './grid';
import RawText from './text';
import RawLogo from './logo';

export const Action = RawAction;
export const Button = RawButton;
export const Header = RawHeader;
export const Subheader = Subhead;
export const Hero = RawHero;
export const Link = RawLink;
export const BlankLink = BLink;
export const Section = RawSection;
export const Table = RawTable;
export const TableHeader = TH;
export const TableBody = TB;
export const VerticalTable = VTable;
export const List = RawList;
export const ListItem = LI;
export const Row = R;
export const Column = C;
export const Text = RawText;
export const Logo = RawLogo;