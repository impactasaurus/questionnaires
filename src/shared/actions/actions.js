import React, { Fragment } from 'react';
import styled from 'styled-components';
import { spacings } from '../theme';

const List = styled.ul`
    padding-top: ${spacings.xl};
`;

export const Line = styled.div`
    flex-grow: 0.5;
    width: 1px;
    border-left: 1px solid ${props => props.theme.fg};
    box-sizing: border-box;
`;

export default ({children}) => (
    <Fragment>
        <List>
            {React.Children.map(children, a => <li>{a}</li>)}
        </List>
        <Line />
    </Fragment> 
);