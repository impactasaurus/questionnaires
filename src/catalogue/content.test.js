import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import createStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { ThemeProvider } from 'styled-components';
import { StaticRouter } from 'react-router-dom';
import fetchMock from 'fetch-mock';
import ConnectedContent, { Content } from './Content';
import { requestCatalogue, receiveCatalogue } from './actions';
import { primaryTheme } from '../shared/theme';
import { Table, TableHeader } from '../shared';

describe('catalogue <Content />', () => {
    const render = (component, initialState) => {
        const store = createStore([ thunk ])(initialState || {});
        return renderer.create(
            <Provider store={store}>
                <StaticRouter context={{}}>
                    <ThemeProvider theme={primaryTheme}>
                        {component}
                    </ThemeProvider>
                </StaticRouter>
            </Provider>
        );
    };

    it('should load the catalogue on mount', () => {
        const catalogueSpy = jest.fn();
        render(<Content loadCatalogue={catalogueSpy} />);
        expect(catalogueSpy).toHaveBeenCalled();
    });

    it('should pass the given data to the table', () => {
        const data = [
            { some: 'data', id: '1' },
        ];
        const component = render(<Content loadCatalogue={jest.fn()} data={data} />);
        const table = component.root.findByType(Table);
        expect(table.props.data).toEqual(data);
    });

    it('should render the table columns', () => {
        const data = [
            {
                id: 1,
                logo: 'logo.png',
                key: 'questionnaire',
                name: 'The questionnaire',
                description: 'questionnaire description',
                length: 10,
                demographic: ['some', 'demographic'],
                sectors: ['some', 'sectors'],
            },
        ];
        const component = render(<Content loadCatalogue={jest.fn()} data={data} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});

describe('catalogue <ConnectedContent />', () => {
    beforeEach(fetchMock.restore);

    const render = (component, store) => renderer.create(
        <Provider store={store}>
            <StaticRouter context={{}}>
                <ThemeProvider theme={primaryTheme}>
                    {component}
                </ThemeProvider>
            </StaticRouter>
        </Provider>
    );

    it('should dispatch the load catalogue action creator', async () => {
        const endpoint = '/api/catalogue.json';
        fetchMock.get(endpoint, { status: 200, body: [{ some: 'catalogue' }] });
        const store = createStore([ thunk ])({});
        
        render(<ConnectedContent />, store);
        await fetchMock.flush();

        expect(store.getActions()).toEqual([
            requestCatalogue(),
            receiveCatalogue([{ some: 'catalogue' }]),
        ]);
        expect(fetchMock.called(endpoint)).toBeTruthy();
    });

    it('should load the data from the catalogue', async () => {
        const endpoint = '/api/catalogue.json';
        fetchMock.get(endpoint, { status: 404 });
        const data = { id: '1', some: 'catalogue' };
        const store = createStore([ thunk ])({
            catalogue: [{ ...data }],
        });
        
        const component = render(<ConnectedContent />, store).root;
        await fetchMock.flush();
        
        expect(component.findByType(Content).props.data).toEqual([{ ...data }]);
    });
});