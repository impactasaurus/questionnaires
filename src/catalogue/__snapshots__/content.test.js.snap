// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`catalogue <Content /> should render all the columns 1`] = `
.c3 {
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  color: rgb(153,153,153);
  line-height: 3.375rem;
  font-size: 2.25rem;
  font-weight: 300;
  margin: 0;
}

.c3::after {
  content: '';
  height: 1px;
  display: block;
  width: 8rem;
  margin-top: 0.75rem;
  border-bottom: 1px solid rgb(153,153,153);
}

.c9 {
  color: rgb(153,153,153);
  border-bottom: 1px dotted rgb(153,153,153);
  text-decoration: none;
  -webkit-transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  transition: color 0.2s ease-in-out,border-color 0.2s ease-in-out;
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  line-height: 1.5rem;
  font-size: 1rem;
}

.c9:hover {
  border-bottom-color: transparent;
  color: rgb(157,198,107);
}

.c0 {
  padding: 1.5rem 2.25rem;
}

.c5 {
  border-bottom: 1px solid rgb(153,153,153);
}

.c7:nth-child(2n) {
  background-color: rgba(157,198,107,0.075);
}

.c4 {
  width: 100%;
}

.c6 {
  color: rgb(153,153,153);
  text-align: left;
  padding: 0 0.75rem 0.75rem 0.75rem;
  font-family: "Source Sans Pro",Helvetica,sans-serif;
}

.c8 {
  padding: 0.75rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  padding-bottom: 1.5rem;
}

.c1 > * {
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
}

.c1::after {
  content: '';
  display: table;
  clear: both;
}

.c2 {
  width: 100%;
}

.c10 {
  color: rgb(153,153,153);
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  font-size: 1rem;
  font-weight: 300;
  line-height: 1.5rem;
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <h1
        className="c3"
      >
        Catalogue
      </h1>
    </div>
  </div>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      <table
        className="c4"
      >
        <thead
          className="c5"
        >
          <tr>
            <th
              className="c6"
            >
              Name
            </th>
            <th
              className="c6"
            >
              Version
            </th>
            <th
              className="c6"
            >
              Description
            </th>
            <th
              className="c6"
            >
              License
            </th>
            <th
              className="c6"
            >
              Attribution
            </th>
            <th
              className="c6"
            >
              Number of questions
            </th>
          </tr>
        </thead>
        <tbody
          className=""
        >
          <tr
            className="c7"
          >
            <td
              className="c8"
            >
              <a
                className="c9"
                href="/questionnaire/questionnaire1"
                onClick={[Function]}
              >
                Questionnaire 1 name
              </a>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                1.0.0
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                Questionnaire 1 desc
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                Questionnaire 1 license
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                Questionnaire 1 attribution
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                3
              </p>
            </td>
          </tr>
          <tr
            className="c7"
          >
            <td
              className="c8"
            >
              <a
                className="c9"
                href="/questionnaire/questionnaire2"
                onClick={[Function]}
              >
                Questionnaire 2 name
              </a>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                1.0.0
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                Questionnaire 2 desc
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                Questionnaire 2 license
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                Questionnaire 2 attribution
              </p>
            </td>
            <td
              className="c8"
            >
              <p
                className="c10"
              >
                2
              </p>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</section>
`;
